File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        sum=0
        best=-2**(31)
        
        for i in nums:
            
            sum=max(sum+i,i)
            
            best=max(best,sum)
            
            
        return best

File: 53-maximum-subarray/53-maximum-subarray.c
# define MAX(a,b) (a<=b?b:a)

int maxSubArray(int* nums, int numsSize){

    int sum=0;
    int best=INT_MIN;
    
    for (int i=0; i<numsSize;i++){
        
        sum=MAX(nums[i]+sum,nums[i]);
        best=MAX(best,sum);
    
    }
    
    return best;
}